doc/SONiC_VRRP_HLD.md

```diff
@@ -0,0 +1,582 @@
+
+configure interface vrrp priority <interface_name> <vrid> <priority>
+This command configures priority for a VRRP instance
+- priority - a number between 1 and 254 with 1 being the lowest and 254 being the highest priority. Default is 100. Priority 255 is reserved for owner VRRP router.
+
+configure interface vrrp adv_interval <interface_name> <vrid> <interval>
+This command configures VRRP periodic advertisement interval for a VRRP instance
+- interval - a number between 1 and 255. Unit is in seconds. Default is 1sec. 
+
+configure interface vrrp pre_empt enable <interface_name> <vrid>
+This command enables pre-emption of a Master when a higher priority VRRP router arrives
+- enable - Enable pre-emption. Default is enabled. 
+
+configure interface vrrp pre_empt disable <interface_name> <vrid>
+This command disables pre-emeption of a Master when a higher priority VRRP router arrives
+- disable - Disable pre-emption. Default is enabled. 
+
+configure interface vrrp track_interface add <interface_name> <vrid> <track_interface> <weight>
+This command adds a track interface to a VRRP Instance. A maximum of 8 track interfaces can be added to a VRRP instance. 
+- track_interface - Interface to track. Interface can be Ethernet, Vlan or PortChannel
+- weight - weight or importance assigned to the track_interface. When track interface goes down, the priority of VRRP instance will be reduced by weight
+
+configure interface vrrp track_interface remove <interface_name> <vrid> <track_interface> 
+This command removes an already configured track interface from a VRRP Instance. 
+```
+
+### 5.2 Show Commands
+
+```
+show vrrp
+ - lists all the VRRP instances including their current state
+ Sample output:-
+admin@sonic:~$ show vrrp
+Interface_Name  VRID   State             VIP  Cfg_Prio  Curr_Prio
+         Vlan1     1  Backup       4.1.1.100       100        120
+         Vlan2     2  Master       4.1.2.100       100        100
+         Vlan3     3  Backup       4.1.3.100       100        100
+         Vlan4     4  Backup       4.1.4.100       100        100
+         Vlan5     5  Master       4.1.5.100       100        100
+
+
+show vrrp <interface_name> <vrid>
+ - This command displays data about a VRRP instance in detail
+ Sample output:-
+admin@sonic:~$ show vrrp Vlan1 1
+Vlan1, VRID 1
+Version is 2
+State is Backup
+Virtual IP address:
+  4.1.1.100
+Virtual MAC address is 0000.5e00.0101
+Track interface:
+  Intfname       State  Priority
+  Ethernet7      Up     10
+  PortChannel001 Up     10
+Configured Priority is 100, Current Priority is 120
+Advertisement interval is 1 sec
+Preemption is enabled
+```
+
+
+## 6 Serviceability and Debug
+
+The existing logging mechanisms shall be used. Proposed debug framework shall be used for internal state dump.
+
+## 7 Warm Reboot Support
+
+Currently, warm-reboot is not supported for VRRP. That is, warm-reboot will simply restart the VRRP docker without VRRP storing any data for warm restart. 
```

Comment:
we should prevent warm reboot if there is vrrp configuraiton in place.

Reply:
In the new design VRRP master's will relinquish the mastership of VRRP instances (by sending priority 0 keepalives, so that the standby can take over the mastership instantaneously) before going to warm-reboot. During warm-reboot the neighboring routers will be master. Post warm-reboot the rebooted router could claim back the mastership.

This way the forwarding is kept functional (and not changing) during the process of warm-reboot.